vowpalwabbit/parser.h

```diff
@@ -34,11 +45,12 @@ struct parser
   example* examples;
   uint64_t used_index;
   bool emptylines_separate_examples; // true if you want to have holdout computed on a per-block basis rather than a per-line basis
-  MUTEX examples_lock;
-  CV example_available;
-  CV example_unused;
-  MUTEX output_lock;
-  CV output_done;
+
+  std::mutex examples_lock;
+  std::condition_variable example_available;
```

Comment:
A comment saying which lock to use with each cond_var goes a long way when reviewing for concurrency.